api/system-backup/read

```diff
@@ -88,44 +88,51 @@ sub get_default_custom_excludes
     return \@tmp;
 }
 
-sub find_latest_log
+sub find_all_logs
 {
     my $name = shift;
-    my $n = "[0123456789]";
-    my $latest_filename = "";
-    my $latest_ts = 0;
+    my $n = "[0-9]";
+    my @logs = ();
     foreach my $file (glob("/var/log/backup/backup-$name-$n$n$n$n$n$n$n$n$n$n$n$n.log")) {
         my $stat = stat($file);
-        if ($stat->mtime > $latest_ts) {
-            $latest_ts = $stat->mtime;
-            $latest_filename = $file
+        if ( -f $file ) {
+            open my $fh, '<', $file;
+            $/ = undef;
+
+            my $path = $file;
+            $file =~ s/[^0-9]//g;
+            my ($y,$M,$d,$h,$m) = $file =~ /^([0-9]{4})([0-9]{2})([0-9]{2})([0-9]{2})([0-9]{2})\z/ or error();
+
+            push(@logs, {"date" => "$y-$M-$d $h:$m", "data" => <$fh>, "path" => $path, "ts" => $stat->mtime});
```

Comment:
Do not slurp file: this API will soon fill the system machine RAM if there are many backup files (eg. a backup scheduled once per hour).

Reply:
To avoid this, we can call asynchronously from the UI the content of the log when the user change the backup date selector.

Reply:
I like the idea!